src/test/java/bio/overture/ego/controller/UserControllerTest.java

```diff
@@ -207,32 +204,28 @@ public void createUser_NonExisting_Success() {
 
     // Create the user
     val user = createUserPostRequestAnd(r).extractOneEntity(User.class);
-    assertThat(user)
-        .isEqualToIgnoringGivenFields(
-            r,
-            ID,
-            USERAPPLICATIONS,
-            USERPERMISSIONS,
-            USERGROUPS,
-            TOKENS,
-            NAME,
-            CREATEDAT,
-            LASTLOGIN);
+    Assert.assertEquals(user.getEmail(), r.getEmail());
+    Assert.assertEquals(user.getFirstName(), r.getFirstName());
+    Assert.assertEquals(user.getLastName(), r.getLastName());
+    Assert.assertEquals(user.getPreferredLanguage(), r.getPreferredLanguage());
+    Assert.assertEquals(user.getType(), r.getType());
+    Assert.assertEquals(user.getStatus(), r.getStatus());
 
     // Assert the user can be read and matches the request data
     val r1 = getUserEntityGetRequestAnd(user).extractOneEntity(User.class);
-    assertThat(r1)
-        .isEqualToIgnoringGivenFields(
-            r,
-            ID,
-            USERAPPLICATIONS,
-            USERPERMISSIONS,
-            USERGROUPS,
-            TOKENS,
-            NAME,
-            CREATEDAT,
-            LASTLOGIN);
-    assertThat(r1).isEqualToIgnoringGivenFields(user);
+    Assert.assertEquals(r1.getEmail(), r.getEmail());
+    Assert.assertEquals(r1.getFirstName(), r.getFirstName());
+    Assert.assertEquals(r1.getLastName(), r.getLastName());
+    Assert.assertEquals(r1.getPreferredLanguage(), r.getPreferredLanguage());
+    Assert.assertEquals(r1.getType(), r.getType());
+    Assert.assertEquals(r1.getStatus(), r.getStatus());
+
+    Assert.assertEquals(r1.getEmail(), user.getEmail());
+    Assert.assertEquals(r1.getFirstName(), user.getFirstName());
+    Assert.assertEquals(r1.getLastName(), user.getLastName());
+    Assert.assertEquals(r1.getPreferredLanguage(), user.getPreferredLanguage());
+    Assert.assertEquals(r1.getType(), user.getType());
+    Assert.assertEquals(r1.getStatus(), user.getStatus());
```

Comment:
The problem with comparing specific fields is that if one is added, you have to remember to update the test.

If that's really an issue, though, we can do something with reflection (if we have to...)

Reply:
True, but right now I believe it to be one of the biggest hits on compile time. What might be good is if we just implement it ourselves at some point on top of JUnit